lib/Drush/Boot/BaseBoot.php

```diff
@@ -62,6 +62,10 @@ function bootstrap_and_dispatch() {
           $command += $this->command_defaults();
           // Insure that we have bootstrapped to a high enough
           // phase for the command prior to enforcing requirements.
+          if ($command['command'] == 'updatedb-batch-process') {
```

Comment:
Shouldn't check `$command['command']` here; instead, invent another property that indicates what kind of Kernel the command needs. Maybe pass that in to `drush_bootstrap_to_phase()`.